hs/src/Reach/Eval/Core.hs

```diff
@@ -5566,11 +5566,9 @@ doParallelReduce lhs (ParallelReduceRec {..}) = locAt slpr_at $ do
   let commit_s = JSMethodCall (jid "commit") a JSLNil a sp
   let while_body = [commit_s, fork_s]
   let while_s = JSWhile a a while_e a $ JSStatementBlock a while_body a sp
-  block_s <- case pr_mdef of
-    Just (JSArrowExpression _ _ b) -> return $ jsArrowBodyToStmt b
-    Just ow -> locAtf (srcloc_jsa "define" $ jsa ow) $ expect_ Err_ParallelReduce_DefineBlock
-    Nothing -> return $ JSStatementBlock a [] a sp
-  let pr_ss = [var_s, block_s, inv_s, while_s]
+  block_ss <- flip concatMapM pr_defs $ fmap snd . deconstructFunStmts
```

Comment:
The root of that error is that `deconstructFunStmts` evals the argument when it is not an arrow/etc